.github/workflows/iroha2-dev-nightly.yml

```diff
@@ -2,13 +2,39 @@ name: I2::Dev::Nightly::Publish
 
 on: workflow_dispatch
 
+env:
+  DOCKER_COMPOSE_PATH: defaults
+
 jobs:
+  build_executor:
```

Comment:
Can we somehow avoid this repetition i.e. can we have the only `build_executor` job somewhere and make the subsequent jobs call it?

Reply:
Yes, but it's not a good decision. We might do it, but from my нexperience it requires to move workflows from PR trigger to `workflow run` trigger that will break some things. And also we might encounter GitHub API rate limit. That is, too significant disadvantages.

Reply:
Ok, I found it is not simple to achieve an equivalent of the combined condition of `on.workflow_run` and `on.pull_request.paths` when we want to take the single build approach.

On the other hand, what if we move the `build_executor` job to a [reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows) e.g. `build-executer.yml` and replace every `build_executor` job with a step that `uses: ./.github/workflows/build-executer.yml`? It would at least save the literal repetition, but I don't know if it hits the API rate limit

Reply:
Considered with Sato that it's not acceptable for our CI case